File: sketch/src/main/java/com/github/panpf/sketch/decode/NotFoundGifSoException.java
package com.github.panpf.sketch.decode;

import androidx.annotation.NonNull;

import com.github.panpf.sketch.SketchException;

public class NotFoundGifSoException extends SketchException {
    public NotFoundGifSoException(@NonNull UnsatisfiedLinkError cause) {
        super(cause);
    }

    public NotFoundGifSoException(@NonNull ExceptionInInitializerError cause) {
        super(cause);
    }

    @NonNull
    @Override
    public synchronized Throwable getCause() {
        return super.getCause();
    }
}

File: sketch/src/main/java/com/github/panpf/sketch/decode/NotFoundGifSoException.kt
package com.github.panpf.sketch.decode

import com.github.panpf.sketch.SketchException

class NotFoundGifSoException : SketchException {
    constructor(cause: UnsatisfiedLinkError) : super(cause) {}
    constructor(cause: ExceptionInInitializerError) : super(cause) {}

    @get:Synchronized
    override val cause: Throwable
        get() = super.cause!!
}